File: 백준/Bronze/1816. 암호 키/암호 키.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; i++) {
            long num = Long.parseLong(br.readLine());

            boolean YES = true;
            if (num % 2 == 0) {
                bw.write("NO\n");
                YES = false;
            } else {
                for (int j = 3; j <= 1_000_000; j += 2) {
                    if (num % j == 0) {
                        bw.write("NO\n");
                        YES = false;
                        break;
                    }
                }
            }

            if (YES) {
                bw.write("YES\n");
            }
        }

        bw.flush();
    }
}

File: 백준/Bronze/1816. 암호 키/암호 키.py
N = int(input())

for _ in range(N):
    num = int(input())
    isTrue = True
    if num % 2 == 0:
        isTrue = False
        print("NO")
    else:
        for i in range(3, 1_000_001, 2):
            if num % i == 0:
                isTrue = False
                print("NO")
                break
    if isTrue:
        print("YES")
